hubtools/src/lib.rs

```diff
@@ -349,6 +333,38 @@ impl RawHubrisImage {
             .copy_from_slice(input.as_bytes());
         Ok(())
     }
+
+    // TODO: Knowlege about image formats needs to be
+    // consolidated into its own crate usable in std and no_std
+    // contexts.
+
+    fn locate_header(&self) -> Result<u32, Error> {
+        for header_offset in header::POSSIBLE_OFFSETS {
+            let mut header_magic = 0u32;
+            self.read(self.start_addr + header_offset, &mut header_magic)?;
+            if header::MAGIC.contains(&header_magic) {
+                return Ok(header_offset);
+            }
+        }
+        Err(Error::MissingMagic(header::MAGIC))
+    }
+
+    fn read_image_header(&self) -> Result<header::ImageHeader, Error> {
+        let header_offset = self.locate_header()?;
+        let mut header = header::ImageHeader::new_zeroed();
+        // TODO: We assume that the host that we're running on matches
+        // the little-endianness of the target device. That happens to
+        // be true in all known cases, but is not good practice and
+        // should be fixed.
+        self.read(self.start_addr + header_offset, &mut header)?;
+        Ok(header)
+    }
+
+    /// Return the image size up to but not including any signature block.
+    pub fn image_size(&self) -> Result<u32, Error> {
+        let header = self.read_image_header()?;
+        Ok(header.total_image_len)
+    }
```

Comment:
What's the plan for this pub function?

Reply:
This doesn't need to be pub right now. Someday, ImageHeader goes away on the RoT and all image format knowledge goes into one crate.
`image_size` on an LPC55 Signed XIP image shouldn't need ImageHeader.